core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala

```diff
@@ -155,6 +155,46 @@ class TopicCommandWithAdminClientTest extends KafkaServerTestHarness with Loggin
     adminClient.listTopics().names().get().contains(testTopicName)
   }
 
+  @Test
+  def testCreateWithDefaults(): Unit = {
+    createAndWaitTopic(new TopicCommandOptions(
+      Array("--topic", testTopicName)))
+
+    adminClient.listTopics().names().get().contains(testTopicName)
+  }
+
+  @Test
+  def testCreateWithDefaultReplication(): Unit = {
+    createAndWaitTopic(new TopicCommandOptions(
+      Array("--topic", testTopicName, "--partitions", "2")))
+
+    val partitions = adminClient
+      .describeTopics(Collections.singletonList(testTopicName))
+      .all()
+      .get()
+      .get(testTopicName)
+      .partitions()
+      .size()
+    assertEquals(partitions, 2)
```

Comment:
Should we be asserting that the default replication factor was used? Similarly for the other two tests.

Reply:
I was worried that the overhead for getting that data is high for this unit test, but I'm happy to add it (my thought was that if you don't specify replicas/partitions you don't really care what it gets set to so long as the topic gets created). I will add it

Reply:
Not sure I follow. We would still only need one call to `describeTopics`.

Reply:
Nevermind... I thought it might require changing a lot of test code to pass in the defaults to the broker config but it's pretty targeted. Otherwise I'd need a call to describe cluster to get the broker ID and then to describe configs to get the default configs.